firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -35,10 +43,24 @@
 public class FirebaseInstallations implements FirebaseInstallationsApi {
 
   private final FirebaseApp firebaseApp;
+  private final FirebaseInstallationServiceClient serviceClient;
+  private final PersistedFid persistedFid;
+  private final Executor executor;
+  private final int THREAD_COUNT = 6;
 
   /** package private constructor. */
   FirebaseInstallations(FirebaseApp firebaseApp) {
+    this(firebaseApp, new PersistedFid(firebaseApp), new FirebaseInstallationServiceClient());
+  }
+
+  FirebaseInstallations(
+      FirebaseApp firebaseApp,
+      PersistedFid persistedFid,
+      FirebaseInstallationServiceClient serviceClient) {
     this.firebaseApp = firebaseApp;
+    this.serviceClient = serviceClient;
+    this.executor = Executors.newFixedThreadPool(THREAD_COUNT);
```

Comment:
a fixed thread pool never GC's idle threads ("The threads in the pool will exist until it is explicitly shutdown.")

1. are you sure we need 6 threads? that seems like a huuuuge amount
2. can we not inject the executor in this constructor? it's usually an antipattern for the instantiating class to also instantiate its own thread pool

Reply:
+1

Reply:
I was referring to FLoC sdk while implementing the executor. I am not sure whats the best approach to decide the thread pool count.

If I just inject the executor, the user of the sdk has to instantiate the executor class. This might be a wrong question, but, how will they know the right number of threads required? Or, how does this improve the performance?

Reply:
do you actually need multiple threads? a cached executor service w/ a cap of one thread and ~30s idle time should be sufficient, no? You're never actually running more than one thing at a time right?

>If I just inject the executor, the user of the sdk has to instantiate the executor class.

I'm suggesting that you move the instantiation of the executor service into the delegating constructor, which ensures there's at least one constructor where you can inject a test-controllable executor.

Reply:
I will be running more than one task at a time( Line 96 and Line 170 ). Task for getId and another task for registering FID. 

Do you recommend using newCachedThreadPool() or have a custom executor

 new ThreadPoolExecutor(0, 2,
                                      30L, TimeUnit.SECONDS,
                                      new SynchronousQueue<Runnable>());

Reply:
You're loading two separate jobs onto the same executor, but they don't run in parallel right? The getId call runs first, then the registration call runs asynchronously after the getId call finishes.

A custom executor is probably fine. `new ThreadPoolExecutor(0, 1, 30L, TimeUnit.SECONDS, new SynchronousQueue<>());`